docker/android-system.sh

```diff
@@ -7,32 +33,39 @@ set -euo pipefail
 . lib.sh
 
 main() {
-    local arch="${1}"
-    local td
-    td="$(mktemp -d)"
-    pushd "${td}"
+    export ARCH="${1}"
+    MAJOR_VERSION=$(echo "${ANDROID_VERSION}" | cut -d '.' -f 1)
+    MINOR_VERSION=$(echo "${ANDROID_VERSION}" | cut -d '.' -f 2)
+    TAG="android-${ANDROID_VERSION}"
 
-    # fake java and javac, it is not necessary for what we build, but the build
-    # script ask for it
-    cat << EOF > /usr/bin/java
-#!/usr/bin/env bash
-echo "java version \"1.7.0\""
-echo "OpenJDK Runtime Environment (IcedTea 2.6.9)"
-echo "OpenJDK 64-Bit Server VM (build 24.131-b00, mixed mode)"
-EOF
+    export MAJOR_VERSION
+    export MINOR_VERSION
+    export TAG
 
-    cat << EOF > /usr/bin/javac
-#!/usr/bin/env bash
-echo "javac 1.7.0"
-EOF
+    if [[ "${ANDROID_SYSTEM_NONE}" == "1" ]]; then
+        rm -rf "${PYTHON_TMPDIR}"
+        rm "${0}"
+        return
+    fi
 
-    chmod +x /usr/bin/java
-    chmod +x /usr/bin/javac
+    if [[ "${ANDROID_SYSTEM_COMPLETE}" != "1" ]] && [[ "${MAJOR_VERSION}" -ge 12 ]]; then
+        echo "Android versions 12 and higher couple APEX tightly into the build system." 1>&2
+        echo "These are currently unsupported, and are unlikely to ever be supported." 1>&2
+        echo "Try using a complete Android system build or disable building Android system." 1>&2
+        echo "Note that a complete Android system build is slow and creates massive images." 1>&2
+        echo "Disabling the Android system build will prevent the use of Android runners." 1>&2
+        echo "If you like support for newer versions, help is always wanted." 1>&2
+        exit 1
+    elif [[ "${MAJOR_VERSION}" -eq 7 ]] && [[ "${MINOR_VERSION}" -eq 1 ]]; then
+        echo "Android version 7.1 is not supported." 1>&2
+        exit 1
+    fi
 
-    # more faking
-    export ANDROID_JAVA_HOME=/tmp
-    mkdir /tmp/lib/
-    touch /tmp/lib/tools.jar
+    local td
+    td="$(mktemp -d)"
+    pushd "${td}"
+
+    fake_java
```

Comment:
Complete Android builds might actually require a true Java installation. I haven't tested this, so YMMV.